server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala

```diff
@@ -99,7 +105,28 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
       }
 
       eventSenders.put(listenerId, clientEventSender)
-      cache.addListener(clientEventSender, filter.orNull, converter.orNull)
+
+      if (includeState) {
+         // If state included, do it async
+         val cf = CompletableFuture.runAsync(new Runnable {
+            override def run(): Unit = {
+               cache.addListener(clientEventSender, filter.orNull, converter.orNull)
+            }
+         }, addListenerExecutor)
+
+         cf.whenComplete(new BiConsumer[Void, Throwable] {
+            override def accept(t: Void, cause: Throwable): Unit = {
+               val resp =
+                  if (cause == null) decoder.createSuccessResponse(h, null)
+                  else decoder.createErrorResponse(h, cause)
+
+               ch.writeAndFlush(resp)
+         }
+         })
+      } else {
+         cache.addListener(clientEventSender, filter.orNull, converter.orNull)
+         ch.writeAndFlush(decoder.createSuccessResponse(h, null))
+      }
```

Comment:
**Minor**:  there are some implicits in place in ```org.infinispan.server```'s ```package.scala``` to convert from java lambdas to scala lambdas. ```Runnable``` and ```BiConsumer``` are not there but could be added:
```scala
import java.util.function.{BiConsumer => J8BiConsumer}

implicit def asJavaBiConsumer[T, U](f: (T, U) => Unit): J8BiConsumer[T, U] = new J8BiConsumer[T, U] {
   override def accept(t: T, u: U): Unit = f(t, u)
}

implicit def asJavaRunnable(f: () => Unit): Runnable = new Runnable {
   override def run(): Unit = f()
}
```

allowing the above to be written as:

```
val cf = CompletableFuture.runAsync(() => cache.addListener(clientEventSender, filter.orNull, converter.orNull), addListenerExecutor)
```

and 

```scala
cf.whenComplete((t: Void, cause: Throwable) => {
      val resp = if (cause == null) decoder.createSuccessResponse(h, null)
               else decoder.createErrorResponse(h, cause)
      ch.writeAndFlush(resp)
      ()
 })
```